### DISCUSSION TITLE
GC overhead limit exceeded

### DISCUSSION THREAD

**Discussion by JHub-Wei** [apache/skywalking#8144] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

After the OAP is started for a while, the instance is abnormal and the error java.lang.OutOfMemoryError: GC overhead limit exceeded is reported. The VM memory is 8 GB. The JVM memory of the Docker container is set to 5 GB, and it could not be fixed.

### What you expected to happen

The OAP can be normally started and written to the ES.

### How to reproduce

Version 8.4.0 used

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
If you want to discuss about OOM, at least you should provide the result memory analysis. Otherwise, what do you expect from the community?

**Reply by wu-sheng** [reply]
OAP server has performance improvement in every release of 8.x, so what version do you use, how you use, what your memory dump looks like are the basic context you need to provide

**Reply by JHub-Wei** [reply]
As I mentioned in my discussion, version 8.4.0 is used.Consume Kafka and write it to ES7.

**Comment by JHub-Wei** [comment]
Two pieces of memory data are dumped every two minutes after startup. The comparison shows that the number of ServiceInstanceRelationClientCpmMetrics instances increases by 64319.

**Reply by wu-sheng** [reply]
`64319` seems pretty normal if you have a complex topology in instance level. Such as A service has 50 pods, and B has another 50 pods, then you simply get 2500 call relationship for just one call between 2 services.

**Reply by JHub-Wei** [reply]
Do you have any suggestions or solutions for this situation? Thank you.

**Reply by wu-sheng** [reply]
First of all, I would like to ask you to upgrade to 8.8.1. The change logs between 8.4.0 and 8.8.1, could show you that we improve a lot about data persistence. You could check whether this could help you. In another hand, if there is no leak(if you have verified GC and low load help), increasing memory to avoid full GC and OOM is important. We have seen users set memory to 20G to keep the performance high enough.

**Reply by JHub-Wei** [reply]
Okay, thank you very much.

**Comment by JHub-Wei** [comment]
![oom1](https://user-images.githubusercontent.com/48902832/142366551-5fdd0c84-cf09-4ee6-b004-4b3dd0ce04bc.png)

**Comment by JHub-Wei** [comment]
![oom2](https://user-images.githubusercontent.com/48902832/142366635-47af93bf-14aa-4b4e-ae0b-00b18e40da6e.png)

**Comment by JHub-Wei** [comment]
![oom3](https://user-images.githubusercontent.com/48902832/142366691-e78e90aa-3f82-480e-8274-768c3ed5b8ab.png)

**Comment by JHub-Wei** [comment]
The instance have been on the gc, after running for two or three minutes.
![dashboard3](https://user-images.githubusercontent.com/48902832/142369557-6494b8ee-a5f7-42c2-a7e8-50c5e3102da0.PNG)
